examples/src/main/java/org/eclipse/rdf4j/examples/TempShaclValidation.java

```diff
@@ -0,0 +1,136 @@
+					String.join("\n", "",
+							"@prefix ex: <http://example.com/ns#> .",
+							"@prefix sh: <http://www.w3.org/ns/shacl#> .",
+							"@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .",
+							"@prefix foaf: <http://xmlns.com/foaf/0.1/>.",
+
+							"ex:PersonShape",
+							"  a sh:NodeShape  ;",
+							"  sh:targetClass foaf:Person ;",
+							"  sh:property ex:PersonMustHaveIntAge, ex:MaxOneAgePerPerson .",
+
+							"ex:PersonMustHaveIntAge ",
+							"  sh:path foaf:age ;",
+							"  sh:datatype xsd:int ;",
+							"  sh:minCount 1 .",
+
+							"ex:MaxOneAgePerPerson ",
+							"  sh:path foaf:age ;",
+							"  sh:severity sh:Warning ;",
+							"  sh:maxCount 1 ."
+					));
+
+			connection.add(shaclRules, "", RDFFormat.TURTLE, RDF4J.SHACL_SHAPE_GRAPH);
+
+		}
+
+		try (RepositoryConnection connection = dataRepo.getConnection()) {
+			StringReader invalidSampleData = new StringReader(
+					String.join("\n", "",
+							"@prefix ex: <http://example.com/ns#> .",
+							"@prefix foaf: <http://xmlns.com/foaf/0.1/>.",
+							"@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .",
+
+							"ex:peter a foaf:Person ;",
+							"  foaf:age 20, \"30\"^^xsd:int  ."
+
+					));
+
+			connection.add(invalidSampleData, "", RDFFormat.TURTLE);
+		}
+
+		ValidationReport validationReport = validate(shapesRepo, dataRepo);
+
+		System.out.println(validationReport);
+	}
+
+	private static ValidationReport validate(SailRepository shapesRepo, SailRepository dataRepo) {
+		try (RepositoryConnection shapesConnection = shapesRepo.getConnection()) {
+			try (ForwardChainingShapeSource shapeSource = new ForwardChainingShapeSource(shapesConnection)) {
+
+				List<ContextWithShapes> shapes = Shape.Factory.getShapes(shapeSource,
+						new Shape.ParseSettings(true, true));
+
+				try (SailRepositoryConnection dataRepoConnection = dataRepo.getConnection()) {
+					return performValidation(shapes, new ConnectionsGroup(dataRepoConnection.getSailConnection(), null,
+							null, null, new Stats(), null, null, true));
+				}
+
+			}
+		}
+	}
+
+	private static ValidationReport performValidation(List<ContextWithShapes> shapes,
+			ConnectionsGroup connectionsGroup) {
+
+		List<ValidationResultIterator> collect = shapes
+				.stream()
+				.flatMap(contextWithShapes -> contextWithShapes
+						.getShapes()
+						.stream()
+						.map(shape -> shape.generatePlans(connectionsGroup,
+								new ValidationSettings(new Resource[] { null }, false, true, false)))
+				)
+				.map(planNode -> {
+					planNode = new SingleCloseablePlanNode(planNode, ValidationExecutionLogger.getInstance(false));
+					return planNode;
+				})
+				.map(planNode -> {
+					try (CloseableIteration<? extends ValidationTuple, SailException> iterator = planNode.iterator()) {
+						return new ValidationResultIterator(iterator, -1);
+					}
+				})
+				.collect(Collectors.toList());
+
+		return new LazyValidationReport(collect, -1);
+
+	}
+
+}
```

Comment:
This is the example. It calls into the existing code for parsing shapes and validating data without moving the data around.

It also has limited support for other severity levels.